task_4_1.py

```diff
@@ -0,0 +1,44 @@
+import requests
+import threading
+import time
+from multiprocessing import Process
+import os
+
+urls = ['https://img1.fonwall.ru/o/fi/grass-field-meadow-prairie.jpeg?auto=compress&fit=crop&w=1280&h=800',
+'https://img1.fonwall.ru/o/ja/landscape-animals-sunset-nature.jpeg?auto=compress&fit=crop&w=1280&h=800',
+'https://img3.fonwall.ru/o/tu/tree-nature-horizon-wilderness.jpeg?auto=compress&fit=crop&w=1280&h=800',
+'https://img3.fonwall.ru/o/kw/nature-black-and-white-adventure-wildlife-ulfa.jpeg?auto=compress&fit=crop&w=1280&h=800',
+'https://img1.fonwall.ru/o/ty/animals-nature-grass-squirrel.jpeg?auto=compress&fit=crop&w=1280&h=800',
+'https://img3.fonwall.ru/o/cj/landscape-nature-grass-wildlife.jpeg?auto=compress&fit=crop&w=1280&h=800']
+
+def download_th(url):
+    for index, url in enumerate(urls):
+        responce = requests.get(url)
+        url = url.split('?')
+        url = url[0].split('/')
+        url = url[-1]
```

Comment:
Давайте попробуем сделать проще
```suggestion
        url, *_ = os.path.basename(url).split('?')
```

Reply:
В остальных модулях можно также